search/localsearch_nqueens.cpp

```diff
@@ -0,0 +1,170 @@
+/*
+ * @file
+ * @brief [Heuristic local Search](https://en.wikipedia.org/wiki/Local_search_(optimization)) algorithm.
+ * @details
+ * The LS algorithm is the basis of many methods used for optimization problems.
+ * Its system is based on the search for neighborhoods that obtain a better solution.
+ * This exercise solves the [n queens](https://en.wikipedia.org/wiki/Eight_queens_puzzle) problem.
+ * The rule of neighborhood applied in this file is first improvement.
+ * @author [dhernandezl](https://github.com/dhernandezl)
+ */
+
+#include <iostream>
+#include <vector>
+#include <algorithm>
+#include <array>
+
+
+/*  \namespace search
+ *  \brief Heuristic local search algorithm.

```

Comment:
Incorrect, this is for the `search` directory functions.
This will also help the auto-documentation system.
```suggestion
 *  \brief Search algorithms
```